src/states/common/base.rs

```diff
@@ -266,27 +302,57 @@ pub trait Base: Display {
             return;
         };
 
-        self.send_message(dst_id, message)
+        self.send_message(dst_id, message);
     }
 
     fn send_message(&mut self, dst_id: &PublicId, message: Message) {
-        let conn_info = match self.peer_map().get_connection_info(dst_id) {
-            Some(conn_info) => conn_info.clone(),
-            None => {
-                error!(
-                    "{} - Failed to send message to {:?} - not connected",
-                    self, dst_id
-                );
-                return;
-            }
-        };
+        self.send_message_to_targets(&[*dst_id], 1, message);
+    }
+
+    fn send_message_to_targets(
+        &mut self,
+        dst_targets: &[PublicId],
+        dg_size: usize,
+        message: Message,
+    ) {
+        let msg_id = self.network_service_mut().next_msg_id();
+        let conn_infos: Vec<_> = dst_targets
+            .iter()
+            .filter_map(|pub_id| self.peer_map().get_connection_info(pub_id).cloned())
+            .collect();
+
+        if conn_infos.len() < dg_size {
+            warn!(
```

Comment:
Shall this be a `log_or_panic`, so that we can debug it during test?

Reply:
I'm not sure if it's worth panicking here even during tests - if this happening always indicates a bug. Not totally against it, though. @jeanphilippeD what is your opinion?

Reply:
I had this message in the follow up PR 
```
WARN 14:47:44.561105308 [routing::states::common::base base.rs:325] Elder(ef4ac7..(111)) Less than dg_size valid targets! dg_size = 1; targets = Filter { iter: Iter([]) }; msg = Hop(HopMessage { content: SignedRoutingMessage { content: RoutingMessage { src: PrefixSection(prefix: Prefix(111)), dst: PrefixSection(prefix: Prefix(110)), content: Merge(64f9c4..2e1dcf) }, sending nodes: Some(SectionInfo(prefix: Prefix(111), members: {PublicId(name: ef4ac7..), PublicId(name: f32c59..), PublicId(name: f56f74..), PublicId(name: f853ad..), PublicId(name: ffa334..)}, prev_hash_len: 1, version: 19)), signatures: ProofSet { [PublicId(name: ef4ac7..), PublicId(name: f56f74..), PublicId(name: f853ad..), PublicId(name: ffa334..)] }, proving_sections: [] }, signature: .. })
```

With the following seed: Some([2044325391, 837965916, 820907250, 664787039])

So if it is always a bug we have a bug. But also, if we panic here we would not be able to push it.

Reply:
Maybe we should investigate what caused this message with that seed and see if it's a legitimate situation, or a bug?

Reply:
I'll create an issue once we merge both yours and mine.
But since it is related to merge I'd prefer avoiding looking at it too deeply for now.